### DISCUSSION TITLE
add maintanier information to the packages

### DISCUSSION THREAD

**Discussion by developer-guy** [wolfi-dev/os#123] [Ideas]
What about adding a maintainer to the packages? We (w/@dentrax) thought that, with the help of that, we could delegate the responsibility of managing the packages to the maintainer. For example, we can create a new field called `maintainers` that could include all the maintainers responsible for managing that package. Of course, there could be an agreement smth like that to apply as a maintainer for the package, idk.

/cc @dlorenc @kaniini

**Comment by kaniini** [comment]
Packages in Wolfi intentionally do not have maintainers, and are considered "community property."  Is there a reason we should change from this policy?

**Reply by kaniini** [reply]
To elaborate on why Wolfi does not have maintainers, traditional distribution model where packages are maintained by designated individuals creates a lot of unnecessary friction (at least in my opinion, concluded from nearly 15 years of Alpine and 5 years of Debian before that).

Designated maintainers frequently get mad at "their" packages being messed with, even if a non-maintainer change is needed in order to unblock other work.  This happens *all the time* in Debian and Alpine.

Because of the above, distributions create Non-Maintainer-Update (NMU) policies, which are intended to unblock developers who have work to do.  But then those policies are controversial, with different developers taking different positions on the NMU policy.  As an example, in Alpine, there are "low-threshold NMU" developers who love NMUs, and then there are developers who hate NMUs and despise the people who do them.  And then sometimes people make a mistake, assuming that a developer will be happy with an NMU, when in reality they hate NMUs, and then a code of conduct violation happens because the designated maintainer gets angry about the NMU.

All of this is avoided by treating packages as community property: anyone can update them at any time to unblock themselves, and there is no expectation of individual package ownership, so there is no reason to care about NMUs or get mad about them.

**Comment by moseshope** [comment]
Right I agree through all comments.
